Image-Capture-POC/Image-Capture-POC/AppCoordinator.swift

```diff
@@ -0,0 +1,40 @@
+//
+//  AppCoordinator.swift
+//  Image-Capture-POC
+//
+//  Created by Jose Luna on 9/26/25.
+//
+
+import UIKit
+
+final class AppCoordinator {
+    private var coordinatorRegister: [AppTransition: Coordinator] = [:]
+    
+    lazy var navigationController: UINavigationController? = {
+        .init()
+    }()
```

Comment:
Cuál sería la razón de usar lazy en el navigationController? 

El AppCoordinator, siempre va a necesitar un UINavigationController, así que no tendría mucho sentido declararlo como opcional. Además, en qué caso un Coordinator no manejaría navegación?

Igual se podría usar un lazy var de esta forma:

```suggestion
    lazy var navigationController: UINavigationController? = .init()
```

Reply:
Completamente de acuerdo. Me he dejado llevar ya que UIKit por defecto es optional pero en ese caso algunas vistas no manejan navegación; tienes razon.